### ISSUE TITLE
Editing cookie exceptions is broken

### ISSUE THREAD

**Issue by vrillusions** [vrillusions/permit-cookies#21]
Related to #20 since now it stores two records, one for `http` and one for `https` it's harder to find the exact record to delete.  See #20 for how I'm guessing the permission lookup works.  Problem is when removing the permission I'm only going by the full the URI.  This will go through without errors but it won't actually remove the exception.

Editing permissions but not removing it does work but not in an ideal fashion.  For example if you have an existing permission for `http://google.com` that blocks cookies but you are currently at `https://www.google.com` then when checking permissions it will see the permission for `http://google.com`.  If you are still on `https://www.google.com` and change the cookie behavior from block to allow then you will now have two records.  `https://www.google.com` that allows cookies and the existing `http://google.com` that blocks cookies.  Permission manager can handle this and will allow cookies for `https://www.google.com` but not for `http://google.com`.  I could see this being used though. For example allow cookies when `https` but not `http` but probably not

**Comment by vrillusions** [comment]
Did some more poking around at this and think i figured out the check better. This was all tested on nightly release v43.  here's how I think the permission checking works:

| current address in url bar | search strings (first one that matches is used) |
| ----- | ----- |
| https://www.google.com | `https://www.google.com`, `https://google.com` |
| https://google.com | `https://google.com` | 
| http://www.google.com | `http://www.google.com`, `http://google.com` |
| http://google.com | `http://google.com` |

To summarize it's aware of the protocol but not of subdomains.  So if you want to allow cookies for `w3.example.com` but not `example.com` you can do that but to allow cookies for both `http` and `https` it requires two rules. One for `https://w3.example.com` and one for `http://w3.example.com`.  Maybe add that as an option?

**Comment by vrillusions** [comment]
so I did some testing on current release version of 40.0 and the matching algorithm is the same.  That is: 

* if you block `www.google.com` but go to `google.com` cookies would still get set.
* If you block `google.com` and go to `www.google.com` it will still be blocked.
* If on `www.google.com` you try to remove the permission that is set for `google.com` it won't work

So that's an existing bug.  Probably not a big issue since most sites force you to either with or without www.  Probably for SEO stuff.  So that limits this to the whole `http` vs `https`

**Comment by vrillusions** [comment]
My current thought:

* new option "Modify permissions for both http and https", default is checked
* if that option is checked then do actions on both protocols.  So it will attempt to delete cookies at both `http://example.com` and `https://example.com`.  When it creates rules it will create them for both `http://example.com` and `https://example.com`.

I enable this by default to be consistent with previous versions where modifying the permissions is assumed to mean modify permissions for that site as a whole regardless if it's http or https.

Visually I'll represent this when setting the permission.  If I'm working on both protocols than the url shown will not have a protocol.  If I see a protocol then I will only work on that one.  This allows modification when setting it, similar to how strip www will strip it in the text box and you could add it back.

I'll somehow need to address this option not being effective until version 42 of firefox though.
